tests/Psl/Dict/UniqueScalarTest.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tests\Dict;
+
+use PHPUnit\Framework\TestCase;
+use Psl\Dict;
+use Psl\Iter;
+use Psl\Vec;
+
+final class UniqueScalarTest extends TestCase
+{
+
+    public function testUniqueScalars(): void
+    {
+        $array   = Vec\fill(10, 'foo');
+        $array[] = 'bar';
+
+        $unique = Dict\unique_scalar($array);
+
+        static::assertCount(2, $unique);
+        static::assertSame('foo', Iter\first($unique));
```

Comment:
we should also add a test case for other Iterators ( use `Iter\Iterator` class ), and IteratorAggregates ( use `Collection\Map` or any other collection )

```suggestion
        static::assertCount(2, $unique);
        static::assertSame(['foo', 'bar'], $unique);
        static::assertSame('foo', Iter\first($unique));
```